Allow inner hyphens in options

mrwest808 opened the pull request:
This update allows options to have inner hyphens, as highlighted in the issue referenced below. I'm not proud of my regex skills, so let me know if you have a better implementation idea or if I should add any more tests.

Closes #9

mrwest808 commented:
Chances are I'm missing something, but I added the `(^|\s)` part to catch `-{1,2}` preceeded by a space or at the start of the string. Other option-related tests fail when I try `/^-{1,2}/g`. To clarify, I'm targeting the following patterns:

```
-f, --foo
^  ^^^

--bar, -b
^^    ^^

-w, --with-hyphen
^  ^^^
```

Happy to add a test for accessing the hyphenated value! 🙂

mrwest808 commented:
Seems like hyphenated names aren't set correctly on the options object, so this definitely needs more work. I'm on a train with a bad connection atm, but I'll take a look shortly.

lukeed commented:
No worries~! I'm actually going to tinker on this right now too because now I'm intrigued. 😉 

Will push any progress to your branch~!

lukeed commented:
@mrwest808 Let me know how this looks to you~! 

I slightly changed the `$.parse` function because if someone did funky/poorly-formatted inputs, it would return bad outputs:

```js
' --foo-bar , -f '
//=> [ 'foo-bar', '', 'f', '' ]

'--foo,   -f'
//=> [ 'foo', '', 'f' ]

// etc...
```

Parsing tests to follow in another commit 👍

mrwest808 commented:
Nice, that looks great!

Before getting off the train I experimented with a similar minor change to the `parse` function, in hopes of making it a bit easier to grasp at a quick glance. The change would look like:

```diff
exports.parse = function (str) {
-	return (str || '').replace(/(^|\s+)-{1,2}/g, '').split(/,|\s+/).filter(Boolean);
+	return (str || '').replace(/(^|\s+)-{1,2}/g, '').split(',').filter(Boolean);
}
```

It's basically just a visual change, but if you want I can merge it. 😄

lukeed commented:
Definitely looks nicer, but it doesn't trim/split any trailing whitespace from each segment, and adding another `map()` would be a little silly.

```js
' --foo ,    -f '
//=> [ 'foo ', 'f ' ]
```

mrwest808 commented:
Agreed 👍

lukeed commented:
Just saw your edit -- `mri` takes pre-trimmed arguments from `process.argv`, so it doesn't need to trim the arguments at runtime. 

This parsing is purely for Sade's own records, used for both the help-output & for assigning argv-arguments to (these) key names.

lukeed commented:
Going to take it as it is, happy with it 😃 

Thank you @mrwest808~!! I had totally forgotten about this issue.

mrwest808 commented:
Glad I could help!

lukeed commented:
I ended up changing the `parse` method yet again 😆 

Using a `RegExp` to `.split()` the string is a lot faster (+1M op/s) ***and*** it preserves the ability to use a space to delimit the two keys. I had forgotten about that feature --- good thing I stopped being lazy & added tests, or else this would have been a breaking change! 🙀
